### ISSUE TITLE
"window is not defined" error in ModalDialog.svelte

### ISSUE THREAD

**Issue by DePasqualeOrg** [umaranis/svelte-lexical#4]
When I try to use one of the demos in SvelteKit, I get the following error, probably because anything that calls `window` needs to be enclosed in `onMount` so that it doesn't run on the server. I tried to fix this myself so that I can create a pull request, but I'm running into several barriers as I try to build the package and test it in my SvelteKit project.

```
window is not defined
ReferenceError: window is not defined
    at ModalDialog.svelte:64:4
    at run (/<redacted>/node_modules/svelte/internal/index.js:22:12)
    at Array.forEach (<anonymous>)
    at run_all (/<redacted>/node_modules/svelte/internal/index.js:28:9)
    at Object.render (/<redacted>/node_modules/svelte/internal/index.js:1767:13)
    at render_response (file:///<redacted>/.svelte-kit/runtime/server/index.js:1201:27)
    at async respond$1 (file:///<redacted>/.svelte-kit/runtime/server/index.js:2867:4)
    at async resolve (file:///<redacted>/.svelte-kit/runtime/server/index.js:3256:11)
    at async respond (file:///<redacted>/.svelte-kit/runtime/server/index.js:3200:20)
    at async file:///<redacted>/node_modules/@sveltejs/kit/dist/chunks/plugin.js:363:24
```

**Comment by DePasqualeOrg** [comment]
Fixed here: https://github.com/umaranis/svelte-lexical/commit/9155c9d677f145f1b685cc6484000749371a32b3

**Comment by umaranis** [comment]
Hi @DePasqualeOrg, are you able to run the editor at all in a svelte-kit project?

I tried loading the editor in a svelte-kit starter project but got errors around HMR.

```
[HMR][Svelte] Unrecoverable HMR error in <Root>: next update will trigger a full reload
logError @ proxy.js:15
Proxy<Root> @ proxy.js:380
initialize @ start.js:775
_hydrate @ start.js:1740
await in _hydrate (async)
start @ start.js:1777
(anonymous) @ (index):1531
Lexical.dev.js:2850 
        
       Uncaught (in promise) Error: Unable to find an active editor state. State helpers or node methods can only be used synchronously during the callback of editor.update() or editorState.read().
    at getActiveEditorState (Lexical.dev.js:2850:13)
    at Object.$getRoot (Lexical.dev.js:424:26)
    at LexicalRichText.dev.js:206:28
    at beginUpdate (Lexical.dev.js:3121:5)
    at updateEditor (Lexical.dev.js:3210:5)
    at LexicalEditor.update (Lexical.dev.js:8234:5)
    at initializeEditor (LexicalRichText.dev.js:205:12)
    at registerRichText (LexicalRichText.dev.js:620:3)
    at RichTextPlugin.svelte? [sm]:8:17
    at run (index.mjs:18:12)
```

**Comment by DePasqualeOrg** [comment]
I was seemingly able to get it running in my project at one point (though all the styling of the UI disappeared), but at another point I also I got the error you mentioned. I think it would be worth getting an example running in SvelteKit, since my understanding is that this will be the main way Svelte apps are developed in the future.

**Comment by DePasqualeOrg** [comment]
I think the problem may be the theme files. I get the following error on building with SvelteKit:

```
[rollup-plugin-dynamic-import-variables] Unexpected token (10:12)
file: /<redacted>/src/routes/lexical/themes/StickyEditorTheme.js:10:12
```

**Comment by umaranis** [comment]
@DePasqualeOrg, finally managed to make it work with svelte-kit. Should be able to share a working project soon.

**Comment by umaranis** [comment]
See #3 for details on the working svelte-kit project.

Could you submit a PR for this bug? @DePasqualeOrg

**Comment by umaranis** [comment]
Fix by #7
